src/hpm.c

```diff
@@ -1,98 +1,153 @@
 #include <stdio.h>
 #include <string.h>
 #include "hpm.h"
-#ifdef DEBUG
-#include <time.h>
-#endif // DEBUG
 
-typedef struct {
-    hpm_enviaSerie_t    enviar;
-    char                idx,buffer[8];
-    hpm_timStarter      timer;
-    bool                bandera;
-} hpm_t;
+#define RXBUFLEN    32  //Tama�o del buffer de recepcion
+typedef enum{ 
+    ReadMeasure,
+    StartMeasure, 
+    StopMeasure,
+    EnableAutosend,
+    DisableAutosend,
+    }comandoEnviar_t;
 
-static hpm_t hpm;
-
-#define iniciatim()   if (hpm.timer) hpm.timer(5,&hpm.bandera);
+static bool updated = false; //Copiado de nmeaCayenne
+static hpm_enviaSerie_t     handlerEnvio;
+static char                 rxBuffer[RXBUFLEN];   //Buffer de recepcion
+static char                 rxIdx;          //Indice del buffer de recepcion
+static bool                 autosend;
+static uint16_t             pm25;
+static uint16_t             pm10;
+static comandoEnviar_t      ultimoComandoEnviado;
 
+static void enviaOrden (char *const orden, char largo);
 
+//si modo es true es auto send
 void InicializacionHPM(hpm_enviaSerie_t enviaSerie) {
-    hpm.enviar = enviaSerie;
-    hpm.idx = 0;
-    hpm.timer = 0;
+    handlerEnvio = enviaSerie;
+    rxIdx = 0;
+    hpmChangeAutosend(false);
+}
+//Copiado de nmeacayenne
+bool NCupdated(void) {//NO SABEMOS COMO CAMBIA
```

Comment:
La variable `updated` (que es el nucleo de funcionamiento de lo que retorna esa función), es actualizada en `HPMinput()`. Adicionalmente les está causando problemas porque otra función (en `nmeaCayenne.h`) la expone con el mismo nombre

```suggestion
bool HPMupdated(void) {
```